### PR TITLE
Upgrade to powsybl dependencies 2024.2.0

### PR DESCRIPTION
**Pull request by FranckLecuyer**
Update to gridsuite/dependencies#70

### CODE REVIEW

File: src/main/java/org/gridsuite/report/server/ReportController.java [hunk 1]
```diff
@@ -41,32 +49,31 @@ public ReportController(ReportService service) {
     @Operation(summary = "Get the elements of a report, its reporters, and their subreporters")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The elements of the report, reporters and subreporters"),
         @ApiResponse(responseCode = "404", description = "The report does not exists")})
-    public ResponseEntity<List<ReportNode>> getReport(@PathVariable("id") UUID id,
+    public ResponseEntity<List<ReportNodeImpl>> getReport(@PathVariable("id") UUID id,
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/report/server/ReportController.java:52]: why do you need the implementation instead of the interface?

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/report/server/ReportController.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/report/server/ReportController.java:52]: Same thing has been done in powsybl-core, in this PR : https://github.com/powsybl/powsybl-core/pull/3011
when they upgrade to ReportNode V2.1 ...

File: src/main/java/org/gridsuite/report/server/ReportController.java [hunk 2]
```diff
@@ -41,32 +49,31 @@ public ReportController(ReportService service) {
...
                                                       @Parameter(description = "Filter on severity levels. Will only return elements with those severities.") @RequestParam(name = "severityLevels", required = false) Set<String> severityLevels,
                                                       @Parameter(description = "Empty report with default name") @RequestParam(name = "defaultName", required = false, defaultValue = "defaultName") String defaultName) {
         try {
             List<ReportNode> subReporters = service.getReport(id, severityLevels, reportNameFilter, reportNameMatchingType).getChildren();
             return subReporters.isEmpty() ?
-                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getEmptyReport(id, defaultName).getChildren()) :
-                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(subReporters);
+                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getEmptyReport(id, defaultName).getChildren().stream().map(r -> (ReportNodeImpl) r).toList()) :
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/report/server/ReportController.java:60]:
```suggestion
                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getEmptyReport(id, defaultName).getChildren().stream().map(ReportNodeImpl::cast).toList()) :
```

File: src/main/java/org/gridsuite/report/server/ReportController.java [hunk 3]
```diff
@@ -41,32 +49,31 @@ public ReportController(ReportService service) {
...
+                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(subReporters.stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/report/server/ReportController.java:61]:
```suggestion
                ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(subReporters.stream().map(ReportNodeImpl::cast).toList());
```

File: src/main/java/org/gridsuite/report/server/ReportController.java [hunk 4]
```diff
@@ -41,32 +49,31 @@ public ReportController(ReportService service) {
...
         } catch (EntityNotFoundException ignored) {
-            return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getEmptyReport(id, defaultName).getChildren());
+            return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getEmptyReport(id, defaultName).getChildren().stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/report/server/ReportController.java:63]:
```suggestion
            return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getEmptyReport(id, defaultName).getChildren().stream().map(ReportNodeImpl::cast).toList());
```

File: src/main/java/org/gridsuite/report/server/ReportController.java [hunk 5]
```diff
@@ -41,32 +49,31 @@ public ReportController(ReportService service) {
...
         @ApiResponse(responseCode = "404", description = "The reporter does not exists")})
-    public ResponseEntity<List<ReportNode>> getSubReport(@PathVariable("id") UUID id,
+    public ResponseEntity<List<ReportNodeImpl>> getSubReport(@PathVariable("id") UUID id,
                                                          @Parameter(description = "Filter on severity levels. Will only return elements with those severities") @RequestParam(name = "severityLevels", required = false) Set<String> severityLevels) {
         try {
             return ResponseEntity.ok()
                     .contentType(MediaType.APPLICATION_JSON)
-                    .body(service.getSubReport(id, severityLevels)
-                            .getChildren());
+                    .body(service.getSubReport(id, severityLevels).getChildren().stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/report/server/ReportController.java:76]:
```suggestion
                    .body(service.getSubReport(id, severityLevels).getChildren().stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/report/server/ReportControllerTest.java [hunk 6]
```diff
@@ -402,8 +407,8 @@ private void insertReport(String reportsId, String content) throws Exception {
     }
 
     private void assertReportListsAreEqualIgnoringIds(MvcResult result, String expectedContent) throws JsonProcessingException, UnsupportedEncodingException {
-        List<ReportNode> expectedReportNodeList = objectMapper.readValue(expectedContent, new TypeReference<>() { });
-        List<ReportNode> actualReportNodeList = objectMapper.readValue(result.getResponse().getContentAsString(), new TypeReference<>() { });
+        List<ReportNodeImpl> expectedReportNodeList = objectMapper.readValue(expectedContent, new TypeReference<>() { });
+        List<ReportNodeImpl> actualReportNodeList = objectMapper.readValue(result.getResponse().getContentAsString(), new TypeReference<>() { });
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/report/server/ReportControllerTest.java:411]:
```suggestion
        List<ReportNode> expectedReportNodeList = objectMapper.readValue(expectedContent, new TypeReference<List<ReportNodeImpl>>() { });
        List<ReportNode> actualReportNodeList = objectMapper.readValue(result.getResponse().getContentAsString(), new TypeReference<List<ReportNodeImpl>>() { });
```

File: src/test/java/org/gridsuite/report/server/utils/TestUtils.java [hunk 7]
```diff
@@ -68,7 +69,7 @@ public static void assertReportsAreEqualIgnoringIds(ReportNode expectedNode, Rep
         }
     }
 
-    public static void assertReportListsAreEqualIgnoringIds(List<ReportNode> expectedNodeList, List<ReportNode> actualNodeList) {
+    public static void assertReportListsAreEqualIgnoringIds(List<ReportNodeImpl> expectedNodeList, List<ReportNodeImpl> actualNodeList) {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/report/server/utils/TestUtils.java:72]:
```suggestion
    public static void assertReportListsAreEqualIgnoringIds(List<ReportNode> expectedNodeList, List<ReportNode> actualNodeList) {
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]
We can minimize change & impact on our code by only changing for Spring & Jackson (de)serialization.

File: src/main/java/org/gridsuite/report/server/ReportController.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/report/server/ReportController.java]: ok

File: src/main/java/org/gridsuite/report/server/ReportController.java (hunk 4 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/report/server/ReportController.java]: ok

File: src/main/java/org/gridsuite/report/server/ReportController.java (hunk 5 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/report/server/ReportController.java]: ok

File: src/main/java/org/gridsuite/report/server/ReportController.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/report/server/ReportController.java]: ok

File: src/test/java/org/gridsuite/report/server/utils/TestUtils.java (hunk 7 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/report/server/utils/TestUtils.java:72]: see above

File: src/test/java/org/gridsuite/report/server/ReportControllerTest.java (hunk 6 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/report/server/ReportControllerTest.java:411]: No. this doesn't compile ...

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**